user-operator/src/main/java/io/strimzi/operator/user/Main.java

```diff
@@ -154,13 +172,30 @@ private static Admin createAdminClient(UserOperatorConfig config, KubernetesClie
     private static MetricsProvider createMetricsProvider()  {
         MeterRegistry registry = new PrometheusMeterRegistry(PrometheusConfig.DEFAULT);
 
-        // Bind JVM metrics
-        new ClassLoaderMetrics().bindTo(registry);
-        new JvmMemoryMetrics().bindTo(registry);
-        new JvmGcMetrics().bindTo(registry);
-        new ProcessorMetrics().bindTo(registry);
-        new JvmThreadMetrics().bindTo(registry);
+        /*
+         * Bind JVM metrics. Handled as stream of Suppliers to circumvent checkstyle
+         * error for Class Data Abstraction Coupling by limiting direct instantiations.
+         *
+         * See https://checkstyle.sourceforge.io/checks/metrics/classdataabstractioncoupling.html
+         */
+        Stream.<Supplier<MeterBinder>>of(
+                ClassLoaderMetrics::new,
+                JvmMemoryMetrics::new,
+                JvmGcMetrics::new,
+                ProcessorMetrics::new,
+                JvmThreadMetrics::new)
+            .forEach(binder -> binder.get().bindTo(registry));
```

Comment:
TBH, I'm not sure I see the value in this. If you think this code is better, you should just use it. But the comment suggests you think it is a bad code and you do it just to avoid some Checkstyle warning. If that is the case, I think you should just use an annotation to suppress the warning. But this current way seems a bit strange since the comment suggests the code is bad and is there just to avoid checkstyle. 

(I personally would stick with the old way, nut I'm fine with the new one as well)

Reply:
I guess my comment has a bit of a negative tone to it :-) I'll revert this and suppress the error :+1: